Sources/SPMTestSupport/PIFTester.swift

```diff
@@ -241,6 +241,15 @@ public final class PIFBuildSettingsTester {
         }
     }
 
+    public subscript(_ key: PIF.BuildSettings.SingleValueSetting, for platform: PIF.BuildSettings.Platform) -> String? {
+        if let value = buildSettings[key, for: platform] {
+            buildSettings[key, for: platform] = nil
+            return value
+        } else {
+            return nil
+        }
+    }
```

Comment:
not sure about this mutating subscript. why does it need to mutate?

Reply:
The subscript simply follows the pattern of the others around it, filling the hole for this particular signature:

- Existing: `subscript(_ key: SingleValueSetting) -> String?`
- Added because missing: `subscript(_ key: SingleValueSetting, for platform: Platform) -> String?`
- Existing: `subscript(_ key: MultipleValueSetting) -> [String]?`
- Existing: `subscript(_ key: MultipleValueSetting, for platform: Platform) -> [String]?`

The tests that call these subscripts appear to expect each call to clear the respective entry. After all expected settings have been checked (removing them as a side effect), the tests then check that the remainder is indeed empty. Left over entries (due to the list having gone stale) were why the test had been failing. Some of the entries I needed to add had this signature, which was not yet supported by the existing subscripts.